internal/tofu/transform_module_expansion.go

```diff
@@ -75,17 +85,26 @@ func (t *ModuleExpansionTransformer) Transform(g *Graph) error {
 	// Modules implicitly depend on their child modules, so connect closers to
 	// other which contain their path.
 	for _, c := range t.closers {
-		for _, d := range t.closers {
-			if len(d.Addr) > len(c.Addr) && c.Addr.Equal(d.Addr[:len(c.Addr)]) {
-				g.Connect(dag.BasicEdge(c, d))
+		// For a closer c with address ["module.foo", "module.bar", "module.baz"], we'll look up:
+		//
+		// t.closers["module.foo"]
+		// t.closers["module.foo.module.bar"]
+		//
+		// And connect the parent module to c.
+		//
+		// We skip i=0 because c.Addr[0:0].String() == "".
```

Comment:
Perhaps we could clarify this note to mention that the root module ("") will not exist in t.closers and can be skipped?

Reply:
Done.